### ISSUE TITLE
abyss 2.1.4 fails to build with GCC 8

### ISSUE THREAD

**Issue by sjackman** [bcgsc/abyss#264]
```
g++-8 -DHAVE_CONFIG_H -I. -I..  -I.. -I../Common -I../DataLayer -I/private/tmp/abyss-20181109-25061-120ayc3/abyss-2.1.4 -isystem/usr/local/opt/boost/include -isystem/usr/local/opt/boost/include/include -isystem/usr/local/opt/open-mpi/include  -isystem/usr/local/opt/google-sparsehash/include  -isystem/private/tmp/abyss-20181109-25061-120ayc3/abyss-2.1.4/boost_1_56_0 -Wall -Wextra -Werror -fopenmp -g -O2 -c -o abyss_bloom-bloom.o `test -f 'bloom.cc' || echo './'`bloom.cc
In file included from ../BloomDBG/BloomIO.h:6,
                 from bloom.cc:23:
../lib/bloomfilter/BloomFilter.hpp: In member function 'void BloomFilter::writeHeader(std::ostream&) const':
../lib/bloomfilter/BloomFilter.hpp:233:10: error: 'char* strncpy(char*, const char*, size_t)' output truncated before terminating nul copying 8 bytes from a string of the same length [-Werror=stringop-truncation]
   strncpy(header.magic, "BlOOMFXX", 8);
```
See https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/33255/version=high_sierra/testReport/junit/brew-test-bot/high_sierra/install_abyss/

I'd suggest trying…
```c++
memcpy(header.magic, "BlOOMFXX", 8);
```

Is the `l` in `BlOOMFXX` intentionally lower case? Are we bumping the magic version for the new Bloom hash function?

I'd like to set up CI to build ABySS with GCC 5 (Ubuntu 16), GCC 7 (Ubuntu 18), and GCC 8 (most recent).

**Comment by benvvalk** [comment]
Thanks, @sjackman.

I didn't make the change the Bloom filter magic number in the 2.1.4 release.  When I looked at the code, I noticed that there is no format version field and that the magic number is never actually checked, so I decided to deal with it later.

The BloomFilter class in question comes from the [btl_bloomfilter](https://github.com/bcgsc/btl_bloomfilter) repo, and is used in ABySS, BBT, LINKS, and perhaps other projects as well.

I'll work on the Bloom filter next week, after I get a chance to talk to @JustinChu.  I want to be careful not to break other people's applications.

**Comment by benvvalk** [comment]
@sjackman hopefully the `strncpy` error is fixed in 195f19bba03fec18d86cd931b34275222ba667fc. My version of gcc-8 (8.1.0) didn't produce the error, but I made the change anyway.

Preparing the new ABySS release now.

**Comment by benvvalk** [comment]
fixed by 97d9018

**Comment by sjackman** [comment]
Thanks, Ben!
